lib/chef/resource/yum_package.rb

```diff
@@ -38,7 +38,7 @@ def arch(arg=nil)
         set_or_return(
           :arch,
           arg,
-          :kind_of => [ String ]
+          :kind_of => [ String, Array ]
```

Comment:
Cool. If it works with yum_package, then I presume it'll actually work now or soon via the resource resolver because it'll get Chef::Resource::YumPackage either way.